core/storage/src/lib.rs

```diff
@@ -34,49 +29,60 @@ pub const COL_BLOCK_INDEX: Option<u32> = Some(4);
 pub const TOTAL_COLUMNS: Option<u32> = Some(5);
 
 /// Provides a way to access Storage and record changes with future commit.
-pub struct StateDbUpdate<'a> {
-    overlay: Box<OverlayedChanges>,
-    _backend: Box<TrieBackend>,
-    ext: StateExt<'a>,
+pub struct StateDbUpdate {
+    state_db: Arc<StateDb>,
+    root: MerkleHash,
+    committed: HashMap<Vec<u8>, Option<Vec<u8>>>,
+    prospective: HashMap<Vec<u8>, Option<Vec<u8>>>,
```

Comment:
Why is value wrapped in an option?

Reply:
Because `None` means that this key must be removed.